lib/associations/belongs-to-many.js

```diff
@@ -356,25 +359,58 @@ BelongsToMany.prototype.injectAttributes = function() {
-    targetAttribute.onDelete = this.through.model.rawAttributes[this.foreignIdentifier].onDelete || this.options.onDelete;
-    targetAttribute.onUpdate = this.through.model.rawAttributes[this.foreignIdentifier].onUpdate || this.options.onUpdate;
+    targetAttribute.onDelete = this.through.model.rawAttributes[this.otherKey].onDelete || this.options.onDelete;
+    targetAttribute.onUpdate = this.through.model.rawAttributes[this.otherKey].onUpdate || this.options.onUpdate;
 
     if (!targetAttribute.onDelete) targetAttribute.onDelete = 'CASCADE';
     if (!targetAttribute.onUpdate) targetAttribute.onUpdate = 'CASCADE';
   }
 
-  this.through.model.rawAttributes[this.identifier] = _.extend(this.through.model.rawAttributes[this.identifier], sourceAttribute);
-  this.through.model.rawAttributes[this.foreignIdentifier] = _.extend(this.through.model.rawAttributes[this.foreignIdentifier], targetAttribute);
+  this.through.model.rawAttributes[this.foreignKey] = _.extend(this.through.model.rawAttributes[this.foreignKey], sourceAttribute);
+  this.through.model.rawAttributes[this.otherKey] = _.extend(this.through.model.rawAttributes[this.otherKey], targetAttribute);
 
-  this.identifierField = this.through.model.rawAttributes[this.identifier].field || this.identifier;
-  this.foreignIdentifierField = this.through.model.rawAttributes[this.foreignIdentifier].field || this.foreignIdentifier;
+  this.identifierField = this.through.model.rawAttributes[this.foreignKey].field || this.foreignKey;
+  this.foreignIdentifierField = this.through.model.rawAttributes[this.otherKey].field || this.otherKey;
 
   if (this.paired && !this.paired.foreignIdentifierField) {
-    this.paired.foreignIdentifierField = this.through.model.rawAttributes[this.paired.foreignIdentifier].field || this.paired.foreignIdentifier;
+    this.paired.foreignIdentifierField = this.through.model.rawAttributes[this.paired.otherKey].field || this.paired.otherKey;
   }
 
   this.through.model.init(this.through.model.modelManager);
 
+  this.toSource = new BelongsTo(this.through.model, this.source, {
+    foreignKey: this.foreignKey
+  });
+  this.manyFromSource = new HasMany(this.source, this.through.model, {
+    foreignKey: this.foreignKey
+  });
+  this.oneFromSource = new HasOne(this.source, this.through.model, {
+    foreignKey: this.foreignKey,
+    as: this.through.model.name
+  });
+
+  this.toTarget = new BelongsTo(this.through.model, this.target, {
+    foreignKey: this.otherKey
+  });
+  this.manyFromTarget = new HasMany(this.target, this.through.model, {
+    foreignKey: this.otherKey
+  });
+  this.oneFromTarget = new HasOne(this.target, this.through.model, {
+    foreignKey: this.otherKey,
+    as: this.through.model.name
+  });
+
+  if (this.paired && this.paired.otherKeyDefault) {
+    this.paired.toTarget = new BelongsTo(this.paired.through.model, this.paired.target, {
```

Comment:
No, it's created on the other association (in case of two BTM relations on the same through model but they aren't defining otherKey only foreignKey)